### PR TITLE
Introduce error_chain error infrastructure

### PR DESCRIPTION
**Pull request by mulkieran**
Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
I'm achieving here the limited goal of introducing error_chain into devicemapper core. I don't think any of this is really wrong and it seems it will always get benefits for this subset of errors.

**Comment by mulkieran** [comment]
@agrover @trgill review?

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
Depends on #194 , since that PR eliminates an error return in ```table_load```. This in turn means that all method in dm.rs that wrap ```do_ioctl``` calls, can fail only on a ```do_ioctl``` failure.

**Comment by mulkieran** [comment]
Sort of depends on #196, as well. Since those ```unwrap/expect``` calls might have been actual error returns.

File: src/result.rs [hunk 1]
```diff
@@ -65,17 +56,12 @@ impl fmt::Display for DmError {
...
         }
     }
 
     fn cause(&self) -> Option<&Error> {
-        match *self {
-            DmError::Dm(_, _) => None,
-            DmError::Io(ref err) => Some(err),
-            DmError::Nix(ref err) => Some(err),
-        }
+        None
```
Comment (agrover) [inline src/result.rs]: If we're just going to return None for cause(), this is the same as the default impl and we can just omit it.

Comment (mulkieran) [reply src/result.rs]: great!

**Pull request merged**